charts/bridgeai-prediction-service/templates/deployment.yaml

```diff
@@ -30,14 +30,83 @@ spec:
       serviceAccountName: {{ include "bridgeai-prediction-service.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- if or .Values.initContainers .Values.initDbMigrate.enable (and (not .Values.postgresql.enabled) .Values.externalDatabase.create) }}
+      initContainers:
+        {{- if and (not .Values.postgresql.enabled) .Values.externalDatabase.create }}
+        - name: database-create
+          image: {{ template "bridgeai-prediction-service.initDbCreate.image" . }}
+          imagePullPolicy: {{ .Values.initDbCreate.image.pullPolicy }}
+          env:
+            - name: POSTGRES_HOST
+              value: {{ template "bridgeai-prediction-service.databaseHost" . }}
+            - name: POSTGRES_PORT
+              value: {{ template "bridgeai-prediction-service.databasePort" . }}
+            - name: POSTGRES_DB
+              value: {{ template "bridgeai-prediction-service.databaseName" . }}
+            - name: POSTGRES_USER
+              value: {{ template "bridgeai-prediction-service.databaseUser" . }}
+            - name: POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "bridgeai-prediction-service.databaseSecretName" . }}
+                  key: {{ include "bridgeai-prediction-service.databaseSecretPasswordKey" . }}
+            - name: DB_POSTGRES_USERNAME
+              value: {{ .Values.externalDatabase.postgresqlPostgresUser }}
+            - name: PGPASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "bridgeai-prediction-service.databaseSecretName" . }}
+                  key: {{ include "bridgeai-prediction-service.databaseSecretPostgresPasswordKey" . }}
+          command:
+            - '/bin/sh'
+            - '-c'
+            - |
+              CONN_FLAGS="-h $POSTGRES_HOST -p $POSTGRES_PORT -U $DB_POSTGRES_USERNAME";
+              if [ "$( psql $CONN_FLAGS -tAc "SELECT 1 FROM pg_database WHERE datname=\$\$$POSTGRES_DB\$\$" )" = '1' ]
+              then
+                echo "Database already exists doing nothing";
+                exit 0;
+              fi;
+              createuser $CONN_FLAGS $POSTGRES_USER;
+              PSQL_ESC=$(echo $POSTGRES_PASSWORD | sed 's/./_/g');
+              psql $CONN_FLAGS -c "ALTER ROLE $POSTGRES_USER WITH PASSWORD \$$$PSQL_ESC\$$$POSTGRES_PASSWORD\$$$PSQL_ESC\$";
+              createdb $CONN_FLAGS -O $POSTGRES_USER $POSTGRES_DB;
+        {{- end }}
+        {{- if .Values.initDbMigrate.enable }}
+        - name: database-migrate
+          image: {{ template "bridgeai-prediction-service.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: ['alembic', 'upgrade', 'head']
```

Comment:
Really this should probably only be `alembic` as the command and we should have upgrade and head as arguments in the `values.yaml`, this way we can perform other migrations if needed without having to alter the chart.